ext/hash/hash.c

```diff
@@ -1395,6 +1399,11 @@ static zend_object *php_hashcontext_clone(zend_object *zobj) {
 	zend_object *znew = php_hashcontext_create(zobj->ce);
 	php_hashcontext_object *newobj = php_hashcontext_from_object(znew);
 
+	if (!oldobj->context) {
+		zend_throw_exception(zend_ce_value_error, "Cannot clone a finalized HashContext", 0);
+		return &newobj->std;
+	}
+
```

Comment:
It's better to move ``znew`` creation (``zend_object *znew = php_hashcontext_create(zobj->ce);``) after this check. Otherwise, this should lead to a leak.

Reply:
Oh, I'm wrong. ``clone_obj`` has to return some object even in case of exception. Please, ignore this.
I don't see other problems.